diff --git a/R/zzz.R b/R/zzz.R
--- a/R/zzz.R
+++ b/R/zzz.R
@@ -22,7 +22,7 @@
 
   path <- Sys.getenv(
     x = "RHDF5LIB_RPATH",
-    unset = system.file("lib", package="Rhdf5lib", mustWork=TRUE)
+    unset = "@hdf5@/lib"
   )
 
   sysname <- Sys.info()['sysname']
@@ -47,7 +47,7 @@
                               sprintf('-L%s -lhdf5 %s', 
                                       patharch, winlibs)
                             }, {
-                              sprintf('"%s/libhdf5.a"%s%s', 
+                              sprintf('-L%s -lhdf5%s%s', 
                                       patharch, .getSzipLoc(patharch), .getDynamicLinks(patharch))
                             }
                      )
@@ -58,7 +58,7 @@
                               sprintf('-L%s -lhdf5_cpp -lhdf5 %s', 
                                       patharch, winlibs)
                             }, {
-                              sprintf('"%s/libhdf5_cpp.a" "%s/libhdf5.a"%s%s',
+                              sprintf('-L%s -lhdf5_cpp -L%s -lhdf5%s%s',
                                       patharch, patharch, .getSzipLoc(patharch), .getDynamicLinks(patharch))
                             }
                      )
@@ -69,7 +69,7 @@
                               sprintf('-L%s -lhdf5_hl -lhdf5 %s', 
                                       patharch, winlibs)
                             }, {
-                              sprintf('"%s/libhdf5_hl.a" "%s/libhdf5.a"%s%s', 
+                              sprintf('-L%s -lhdf5_hl -L%s -lhdf5%s%s', 
                                       patharch, patharch, .getSzipLoc(patharch), .getDynamicLinks(patharch))
                             }
                      )
@@ -80,7 +80,7 @@
                               sprintf('-L%s -lhdf5_hl_cpp -lhdf5_hl -lhdf5_cpp -lhdf5 %s', 
                                       patharch, winlibs)
                             }, {
-                              sprintf('"%s/libhdf5_hl_cpp.a" "%s/libhdf5_hl.a" "%s/libhdf5_cpp.a" "%s/libhdf5.a"%s%s',
+                              sprintf('-L%s -lhdf5_hl_cpp -L%s -lhdf5_hl -L%s -lhdf5_cpp -L%s -lhdf5%s%s',
                                       patharch, patharch, patharch, patharch, .getSzipLoc(patharch), .getDynamicLinks(patharch))
                             }
                      )
@@ -102,7 +102,7 @@
 #' getHdf5Version()
 #' @export
 getHdf5Version <- function() {
-  settings_file <- system.file("lib", "libhdf5.settings", package="Rhdf5lib", mustWork=TRUE)
+  settings_file <- "@hdf5@/lib/libhdf5.settings"
   libhdf5_settings <- readLines(settings_file)
   line <- grep("HDF5 Version:", x = libhdf5_settings)
   strsplit(libhdf5_settings[line], split = ": ")[[1]][2]
@@ -113,7 +113,7 @@
 #' @noRd
 #' @keywords internal
 .getDynamicLinks <- function(path) {
-  settings_file <- file.path(path, "libhdf5.settings")
+  settings_file <- "@hdf5@/lib/libhdf5.settings"
   libhdf5_settings <- readLines(settings_file)
   libstr <- grep("Extra libraries", x = libhdf5_settings, fixed = TRUE, value = TRUE) |> 
     gsub("\\s*Extra libraries: ", "", x = _)
